Device expiry implementation

skizzofly opened the pull request:
This modification, permits a device expiration if after one year the record into the OGN DB is not updated.
To renew another year the owner has to go under update, click the check box that he owns the device and save, this will lock the device for the next 365 days.
The expiration status is shown on the device list table, green check is still valid, red X is expired and can be transferred.
Once expired, the device is free to be transferred to an other owner, this to automate device transferring due to inactive accounts
The API's are untouched, a expired device will still show up.

This modification should also loose some work of the admin team, if a flarm owner is really dead or sold the glider, the new owner has to wait for expiry, he will not know if it expired or not, but he can time to time try to transfer the device, if it's denied, means device is still in force of the old user.
![ogn_mod](https://github.com/glidernet/ogn-ddb/assets/49754801/259eca7d-d5e8-4855-b40e-6f680a204aeb)
[database_schema_upgrade.sql.txt](https://github.com/glidernet/ogn-ddb/files/14527289/database_schema_upgrade.sql.txt)
[INSTALL.md](https://github.com/glidernet/ogn-ddb/files/14527393/INSTALL.md)

snip commented on index.php:
```diff
@@ -156,8 +156,10 @@ function changepassword()
+    $ttime = time() -31536000; 
+    $req2 = $dbh->prepare('SELECT * , ( :ti >= dev_updatetime) as dev_valid FROM devices,aircrafts where dev_userid=:us AND dev_actype=ac_id ORDER BY dev_id ASC');
     $req2->bindParam(':us', $_SESSION['user']);
+	$req2->bindParam(':ti', $ttime);
```
Fix indentation here.

snip commented on index.php:
```diff
@@ -715,18 +717,27 @@ function devicelist()
         }        // if owned by the user then update
         else {
-            $error = $lang['error_devexists'];
+            $ttime=time()-31536000; //1 year;
```
Can you please try to have this as a constant and not repeat the same value multiple time?
You can also say this is 365 * 24 * 60 * 60 instead of putting this number a bit unreadable.

snip commented:
<img width="138" alt="image" src="https://github.com/glidernet/ogn-ddb/assets/466133/01a43692-1dbc-4f23-b19d-ca67283b1dd5">
A cross means it is expired or not?
This seems not very straightforward in logic.

skizzofly commented:
Well, a bad red cross means expired (red cross BAD  / green check OK), seems quite logic.  to me :)  
The column is named Not Expired, if it's Green it's not expired, if it's red , it's expired.
Can evaluate if call the column Live or if you prefer, but Unexpired is not a nice label, Not expired is quite explicit

snip commented:
> Well, a bad red cross means expired (red cross BAD / green check OK), seems quite logic. to me :) The column is named Not Expired, if it's Green it's not expired, if it's red , it's expired. Can evaluate if call the column Live or if you prefer, but Unexpired is not a nice label, Not expired is quite explicit

What i see, it is is not clear at all. Users are going to be completly lost with this.
 More over it is not an option of the configuration like the other chek/cross but a status. So it should be more at the begining of the line i think.
Maybe we need to find other icons ...

skizzofly commented:
So tell me how you would implement  this view..

On Thu, Mar 7, 2024 at 7:07 PM Sebastien Chaumontet <
***@***.***> wrote:

> Well, a bad red cross means expired (red cross BAD / green check OK),
> seems quite logic. to me :) The column is named Not Expired, if it's Green
> it's not expired, if it's red , it's expired. Can evaluate if call the
> column Live or if you prefer, but Unexpired is not a nice label, Not
> expired is quite explicit
>
> What i see, it is is not clear at all. Users are going to be completly
> lost with this.
> More over it is not an option of the configuration like the other
> chek/cross but a status. So it should be more at the begining of the line i
> think.
> Maybe we need to find other icons ...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glidernet/ogn-ddb/pull/57#issuecomment-1984141556>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL3TFMKRTTJQQ4YV5OZLV2LYXCUH3AVCNFSM6AAAAABELKUZFCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBUGE2DCNJVGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

snip commented:
> So tell me how you would implement this view..
> […](#)
> On Thu, Mar 7, 2024 at 7:07 PM Sebastien Chaumontet < ***@***.***> wrote: Well, a bad red cross means expired (red cross BAD / green check OK), seems quite logic. to me :) The column is named Not Expired, if it's Green it's not expired, if it's red , it's expired. Can evaluate if call the column Live or if you prefer, but Unexpired is not a nice label, Not expired is quite explicit What i see, it is is not clear at all. Users are going to be completly lost with this. More over it is not an option of the configuration like the other chek/cross but a status. So it should be more at the begining of the line i think. Maybe we need to find other icons ... — Reply to this email directly, view it on GitHub <[#57 (comment)](https://github.com/glidernet/ogn-ddb/pull/57#issuecomment-1984141556)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AL3TFMKRTTJQQ4YV5OZLV2LYXCUH3AVCNFSM6AAAAABELKUZFCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBUGE2DCNJVGY> . You are receiving this because you authored the thread.Message ID: ***@***.***>

Some ideas of icons: https://www.google.com/search?q=expired&tbm=isch&chips=q:expired,g_1:icon:t4tNJzcaTj4%3D&rlz=1C5CHFA_enFR914FR914&hl=en&sa=X&ved=2ahUKEwjt49aX3uKEAxXwuCcCHSDNBlwQ4lYoBnoECAEQPQ&biw=1680&bih=963
But we need something free to use.

We need to agrre on something usable before implementing it. So feel free to propose options.

skizzofly commented:
This states free and seems ok 
License: All icons on this site can be used in personal, commercial, and client projects without any attribution or credit. [License page](https://uxwing.com/license/)

Valid not expired
https://uxwing.com/on-time-icon/

Expired icon
https://uxwing.com/delay-icon/

![intime](https://github.com/glidernet/ogn-ddb/assets/49754801/75df75fa-77af-4a10-8eda-8b84b1d2726f)
![expired](https://github.com/glidernet/ogn-ddb/assets/49754801/8113c30a-f034-41c7-b9ac-3e5505fb1df4)

They are SVG, so they can be big or small.. i resized them to 20x16

snip commented:
> This states free and seems ok License: All icons on this site can be used in personal, commercial, and client projects without any attribution or credit. [License page](https://uxwing.com/license/)
> 
> Valid not expired https://uxwing.com/on-time-icon/
> 
> Expired icon https://uxwing.com/delay-icon/
> 
> ![intime](https://private-user-images.githubusercontent.com/49754801/311000529-75df75fa-77af-4a10-8eda-8b84b1d2726f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4Mzc0MTIsIm5iZiI6MTcwOTgzNzExMiwicGF0aCI6Ii80OTc1NDgwMS8zMTEwMDA1MjktNzVkZjc1ZmEtNzdhZi00YTEwLThlZGEtOGI4NGIxZDI3MjZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA3VDE4NDUxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhYzMyYzU1OGIwMDJhYjRlOGZhOGM0ZjkxNmMyZTdjMmNiYWJmY2M3NWE0ZDAxMzRkMDA2YzRhYjFiODk4YTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GY3ooNTP4rboHjce9pZgWQYtWqKCKMcqkco2YnW2wSE) ![expired](https://private-user-images.githubusercontent.com/49754801/311000534-8113c30a-f034-41c7-b9ac-3e5505fb1df4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4Mzc0MTIsIm5iZiI6MTcwOTgzNzExMiwicGF0aCI6Ii80OTc1NDgwMS8zMTEwMDA1MzQtODExM2MzMGEtZjAzNC00MWM3LWI5YWMtM2U1NTA1ZmIxZGY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA3VDE4NDUxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhN2JkMDBkODE0MWQzMTM1NTdiMjZmZTYzNDUzMDE1MzkwZGRmZDlmMWI0OTBiZTE0ZjcwOTY1ZGU5MzRmM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ATviGUbVAodyW8ApdCSuY34o7SKgYznN0L9hxLzzP2c)
> 
> They are SVG, so they can be big or small.. i resized them to 20x16

Looks nice.
So what should be the column title? "Expired" ?
I think we should also implement a tooltip info when mouse is over to explain what it is.

skizzofly commented:
Regarding  column title  more than Expired, I think Expiration could be
more explicative, Also Expiration Status could be a good candidate.
The tooltip is not a problem, as it's done adding a title to the icon.
Let me know.


On Thu, Mar 7, 2024 at 11:14 PM Sebastien Chaumontet <
***@***.***> wrote:

> This states free and seems ok License: All icons on this site can be used
> in personal, commercial, and client projects without any attribution or
> credit. License page <https://uxwing.com/license/>
>
> Valid not expired https://uxwing.com/on-time-icon/
>
> Expired icon https://uxwing.com/delay-icon/
>
> [image: intime]
> <https://private-user-images.githubusercontent.com/49754801/311000529-75df75fa-77af-4a10-8eda-8b84b1d2726f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4Mzc0MTIsIm5iZiI6MTcwOTgzNzExMiwicGF0aCI6Ii80OTc1NDgwMS8zMTEwMDA1MjktNzVkZjc1ZmEtNzdhZi00YTEwLThlZGEtOGI4NGIxZDI3MjZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA3VDE4NDUxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhYzMyYzU1OGIwMDJhYjRlOGZhOGM0ZjkxNmMyZTdjMmNiYWJmY2M3NWE0ZDAxMzRkMDA2YzRhYjFiODk4YTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GY3ooNTP4rboHjce9pZgWQYtWqKCKMcqkco2YnW2wSE> [image:
> expired]
> <https://private-user-images.githubusercontent.com/49754801/311000534-8113c30a-f034-41c7-b9ac-3e5505fb1df4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4Mzc0MTIsIm5iZiI6MTcwOTgzNzExMiwicGF0aCI6Ii80OTc1NDgwMS8zMTEwMDA1MzQtODExM2MzMGEtZjAzNC00MWM3LWI5YWMtM2U1NTA1ZmIxZGY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA3VDE4NDUxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhN2JkMDBkODE0MWQzMTM1NTdiMjZmZTYzNDUzMDE1MzkwZGRmZDlmMWI0OTBiZTE0ZjcwOTY1ZGU5MzRmM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ATviGUbVAodyW8ApdCSuY34o7SKgYznN0L9hxLzzP2c>
>
> They are SVG, so they can be big or small.. i resized them to 20x16
>
> Looks nice.
> So what should be the column title? "Expired" ?
> I think we should also implement a tooltip info when mouse is over to
> explain what it is.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glidernet/ogn-ddb/pull/57#issuecomment-1984604387>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL3TFMN5DNJAIDAM3GKGGYDYXDREDAVCNFSM6AAAAABELKUZFCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBUGYYDIMZYG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

snip commented:
Hi @skizzofly do you have time to complete updates of this PR?

skizzofly commented:
Hi, I was waiting for a confirmation for the title and the icons ..
Is Expiration and the two icons i posted ok?
Let me know


On Sat, Mar 9, 2024 at 8:07 PM Sebastien Chaumontet <
***@***.***> wrote:

> Hi @skizzofly <https://github.com/skizzofly> do you have time to complete
> updates of this PR?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glidernet/ogn-ddb/pull/57#issuecomment-1986950865>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL3TFMOZRKFG7D6AQL2VOEDYXNMXVAVCNFSM6AAAAABELKUZFCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBWHE2TAOBWGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

snip commented:
"Expiration" and the 2 icons looks ok to me yes.

skizzofly commented:
Ok I also added the tooltip for the 2 cases
Valid is ok  for a "non expired" device, 
Or you preferer a more simple tooltip, for both cases

![ogn_exp_screen](https://github.com/glidernet/ogn-ddb/assets/49754801/f7fdea17-f101-4d60-8b8b-6fcde453a457)

skizzofly commented:
Hi Seb, I just committed the cosmetic fixes with  bef76e4
<https://github.com/glidernet/ogn-ddb/pull/57/commits/bef76e48be72ce8a25f0cdb9ce3bc9fe34f070f7>
1) Changed column title as requested,
2) New icons
3) A static tooltip text, translated in the available languages when
mosueover the icon.

On Sat, Mar 9, 2024 at 8:25 PM Sebastien Chaumontet <
***@***.***> wrote:

> "Expiration" and the 2 icons looks ok to me yes.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glidernet/ogn-ddb/pull/57#issuecomment-1986955337>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL3TFMMPY5KLRPNU3GOK6BDYXNO3JAVCNFSM6AAAAABELKUZFCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBWHE2TKMZTG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

snip commented:
Can you please move this column between the edit and trash button as it is not parameters?

Please also fix other comments.

snip commented:
Instead of "The current device is: Valid" we can say: "This device will expire in x days". What do you think?

skizzofly commented:
the "Will expire in x days tooltip " is in my opinion too much work for a
useless information,
It requires math (conversion from seconds to days) that in the long term
could overload the server for a non so vital information..
IF it's absolutely a vital feature and required for the release i can think
about it.
I Will move the column with no problems

On Sat, Mar 9, 2024 at 11:49 PM Sebastien Chaumontet <
***@***.***> wrote:

> Instead of "The current device is: Valid" we can say: "This device will
> expire in x days". What do you think?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glidernet/ogn-ddb/pull/57#issuecomment-1987000744>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL3TFMMNHPD3CTKIPF4DLJLYXOGXFAVCNFSM6AAAAABELKUZFCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBXGAYDANZUGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

skizzofly commented:
Here is the new layout with the moved column
![ogn_mod1](https://github.com/glidernet/ogn-ddb/assets/49754801/67533b2d-f1b4-442a-b873-848140598466)
Ok it's ok i will commit it.

snip commented:
It looks nice.

snip commented:
About the expiration date i think it needs to be known by the user. So without calculating the expiration date can be write in the tooltip.

skizzofly commented:
This is ok?
![ogn_exp_date](https://github.com/glidernet/ogn-ddb/assets/49754801/f1bd0894-74e0-4764-9a0c-031c4b9d5869)

snip commented:
Maybe a sentence?
"This device registration is going to expire the 7th of March 2025"
(with month in letters to prevent US/EU month/day inversion confusion in date).

Edit: added **registration** keyword

snip commented:
A very nice thing would be to be able to click on the device to renew the expiration for one year.
(so in the tooltip we can add a message "click here to renew it for one year").
But maybe not so easy to implement.

At least we need some explicit message somewhere to explain how to renew the device. And also explain what is the risk of device expired.

skizzofly commented:
I think now it's ok

![ogn_date_expiry](https://github.com/glidernet/ogn-ddb/assets/49754801/baeefbfa-f431-47e4-a5b7-bf1e2cb4c144)

Regarding the update, it's correct **that the user goes in Edit**, reviews the info, and **more important** flags the 
"I certify to be the owner of this device" to save back the info.

Yes we can write a disclamier someware.

Apart these minor cosmetic issues, did you give an eye to the logic?

snip commented on language/english.php:
```diff
@@ -59,9 +59,11 @@
     "table_update" => "Update",
     "table_delete" => "Delete",
     "table_new" => "Add new",
+    "table_tt_expiration" =>"Expiration Status: Green Valid, Red Expired",
```
Please add space before the second string as for the others

snip commented:
> Apart these minor cosmetic issues, did you give an eye to the logic?

Qucikly but you need to comit you change, fix other reported small issues, then squash and finally push.
Then we can get a better view.

skizzofly commented:
Good Sunday Seb,
Check if this disclaimer is ok.. if not, please suggest a adeguate text.
I also added the registration  in the tooltip.
If all is ok  I will commit  and figure out how to squash without making a mess.

![ogn_date_expiry_disclaimer](https://github.com/glidernet/ogn-ddb/assets/49754801/d1187a81-cc2b-4834-8502-fab527a26ee8)

snip commented:
You are going to the good way i think :)

Maybe adding a general information message at the bottom.
And if there is expired device in the list maybe add at top a warning message in red saying that and with quick explanation.

skizzofly commented:
Good evening,
I think now it's ok
If a User ha no expired devices, it will compare as usual.. (first pic)
If a User has and expired device, when a red WARNING will apper, and on the bottom a text message explaining how to renew the device. (second pic)
Check if the text is correct, if not please write me what is correct to you to appear.
Ale.

![ogn_expired_or_not](https://github.com/glidernet/ogn-ddb/assets/49754801/ca2d3515-6207-4af9-9eac-71316513e678)

snip commented:
When you speak about device expired, can you please replace it by device **registration** expired?

Can you please keep the sentence at the bottom of the table in all cases? Can you please remove the Warning for it?

skizzofly commented:
Ok, now i really hope it's all ok
![ogn_date_expiry_new](https://github.com/glidernet/ogn-ddb/assets/49754801/ceeddbc3-428c-4dfa-8c96-88d42c1f0e4d)

snip commented:
We are closed to perfection! Thanks

Few adjustments:
Instead of "one other year" -> "an other year".
Instead of "click on Update, ..." maybe we can say "edit and validate your device".
Is it expected to not have the warning text centered?

Again, please comit you changes, fix other reported small issues, then squash and finally push.

skizzofly commented:
If it's ok i will commit the changes
Thanks.

![ogn_date_expiry_new1](https://github.com/glidernet/ogn-ddb/assets/49754801/bab70c8a-7270-4d75-a3cf-7e12856fdd54)

snip commented:
Looks fine to me.
Thanks.

skizzofly commented:
Ok, Seb,
I committed and squashed the changes as requested.
Going to sleep, tomorrow work calls.
Good night.

snip commented:
@skizzofly please do not forget to fix the small reportes issues at the begining of this thread:
See this screenshot:
<img width="856" alt="image" src="https://github.com/glidernet/ogn-ddb/assets/466133/0e4f6431-0019-4dda-bc23-b270a74ab6cc">
...

skizzofly commented:
Sorry, for some reason i did not see the small reported issues at the begining of this thread..
I made wat was requested, fixed the indentation and removed the repeated value.

snip commented on index.php:
```diff
@@ -735,17 +748,25 @@ function devicelist()
+                $ins = $dbh->prepare('UPDATE devices SET dev_type=:dt, dev_actype=:ty, dev_acreg=:re, dev_accn=:cn, dev_notrack=:nt, dev_noident=:ni, dev_updatetime=:ti, dev_userid=:us WHERE dev_id=:de');
+                //Transfer expired device            
+            }
+            else {
```
Please put `} else { //Transfer expired device` on a single line.

snip commented on index.php:
```diff
@@ -735,17 +748,25 @@ function devicelist()
+            }
         } else {
-            $ins = $dbh->prepare('INSERT INTO devices (dev_id, dev_type, dev_actype, dev_acreg, dev_accn, dev_userid, dev_notrack, dev_noident) VALUES (:de, :dt, :ty, :re, :cn, :us, :nt, :ni)');
+            $ins = $dbh->prepare('INSERT INTO devices (dev_id, dev_type, dev_actype, dev_acreg, dev_accn, dev_userid, dev_notrack, dev_noident,dev_updatetime) VALUES (:de, :dt, :ty, :re, :cn, :us, :nt, :ni, :ti)');
```
sapece missing before dev_updatetime

snip commented on language/czech.php:
```diff
@@ -59,9 +59,13 @@
     "table_new" => "Přidat nové",
+    "table_tt_expiration" =>"Platnost registrace tohoto zařízení vyprší:",
+    "expiry_disclaimer" =>"Zařízení s vypršenou registrací si mohou ostatní uživatelé zdarma zaregistrovat, obnovit zařízení s prošlou platností na další rok, upravit a ověřit své zařízení, zkontrolovat data, zkontrolovat certifikaci vlastnictví zařízení a stisknout Odeslat",
+    "expiry_warning" =>"Upozornění: Platnost registrace některých vašich zařízení vypršela, zkontrolujte je",
```
Sapce missing before `"Upozornění`

snip commented on language/dutch.php:
```diff
@@ -58,10 +58,14 @@
     "table_new" => "Toevoegen",
+    "table_tt_expiration" =>"Deze apparaatregistratie verloopt na de volgende periode:",
+    "expiry_disclaimer" =>"Apparaten met een verlopen registratie kunnen gratis door andere gebruikers worden geregistreerd, een verlopen apparaat met nog een jaar verlengen, uw apparaat bewerken en valideren, de gegevens bekijken, de eigendomscertificering van het apparaat controleren en op Verzenden drukken",
+    "expiry_warning" =>"Waarschuwing: sommige van uw apparaatregistraties zijn VERLOPEN. Controleer ze alstublieft",
```
again sapce missing

snip commented on language/english.php:
```diff
@@ -59,9 +59,13 @@
     "table_new" => "Add new",
+    "table_tt_expiration" =>"This device registration is going to expire the:",
+    "expiry_disclaimer" =>"Devices with expired registration are free to be registered by other users, to renew for another year an expired device, edit and validate your device, review the data, check the device ownership certification and press submit",
+    "expiry_warning" =>"Warning: Some of your devices registration are EXPIRED, please review them",
```
again

snip commented on language/french.php:
```diff
@@ -59,9 +59,13 @@
     "table_new" => "Ajouter",
+    "table_tt_expiration" =>"L'enregistrement de cet appareil va expirer :",
+    "expiry_disclaimer" =>"Les appareils dont l'enregistrement a expiré peuvent être enregistrés librement par d'autres utilisateurs, renouveler pour une année supplémentaire un appareil expiré, modifier et valider votre appareil, examiner les données, vérifier la certification de propriété de l'appareil et appuyer sur Soumettre.",
+    "expiry_warning" =>"Avertissement : L'enregistrement de certains de vos appareils est EXPIRÉ, veuillez les vérifier",
```
again

snip commented on language/german.php:
```diff
@@ -79,9 +79,13 @@
     "table_new" => "Hinzufügen",
+    "table_tt_expiration" =>"Diese Geräteregistrierung läuft ab:",
+    "expiry_disclaimer" =>"Geräte mit abgelaufener Registrierung können von anderen Benutzern kostenlos registriert werden, ein abgelaufenes Gerät um ein weiteres Jahr verlängern, Ihr Gerät bearbeiten und validieren, die Daten überprüfen, die Eigentumsbescheinigung des Geräts überprüfen und auf Senden klicken",
+    "expiry_warning" =>"Warnung: Einige Ihrer Geräteregistrierungen sind ABGELAUFEN. Bitte überprüfen Sie sie",
```
again

snip commented on templates/devicelist.html.twig:
```diff
@@ -29,13 +32,16 @@
                     <img src="/pict/mod.gif" title="{{ lang.table_update }}" class="devbutton">
                 </a>
             </td>
+            <TD class="tab2"><iMG SRC="/pict/ex{{ device.dev_expired }}.png" title="{{ lang.table_tt_expiration }} {{ (device.dev_updatetime+expirationdelta)|date('j F, Y') }} "></TD>
```
Please write TD il lower case as all others

snip commented:
Any chances to fix reported issues?

skizzofly commented:
Seb, i am sorry but i really do not see the repeorted issues.. 
do not see them  in the comments of the changed files and not in the Review section
Regarding the indentation and time constant issues i commited and squased  in to them in to 979f9a0 3 days ago.
Pleas tell me where You reported the issues..

skizzofly commented:
Did you check the fixes?

On Thu, Mar 14, 2024 at 12:04 AM Sebastien Chaumontet <
***@***.***> wrote:

> Any chances to fix reported issues?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glidernet/ogn-ddb/pull/57#issuecomment-1996053476>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL3TFMMFTKO6KGVMYL4UQZTYYDLPFAVCNFSM6AAAAABELKUZFCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOJWGA2TGNBXGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

snip commented:
You did not commit anything since my last message / last check.
See the begingin of this conversation for review part. https://github.com/glidernet/ogn-ddb/pull/57

skizzofly commented:
There is no review started..
Did you submit it?

Il ven 15 mar 2024, 18:51 Sebastien Chaumontet ***@***.***>
ha scritto:

> You did not commit anything since my last message / last check.
> See the begingin of this conversation for review part. #57
> <https://github.com/glidernet/ogn-ddb/pull/57>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glidernet/ogn-ddb/pull/57#issuecomment-2000159464>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL3TFMJGFWU2ZYG2HQFHDJTYYMYLXAVCNFSM6AAAAABELKUZFCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBQGE2TSNBWGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

skizzofly commented:
Ok NOW i see all

Il ven 15 mar 2024, 18:59 Sebastien Chaumontet ***@***.***>
ha scritto:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In index.php
> <https://github.com/glidernet/ogn-ddb/pull/57#discussion_r1516526035>:
>
> >      $req2->bindParam(':us', $_SESSION['user']);
> +	$req2->bindParam(':ti', $ttime);
>
> Fix indentation here.
> ------------------------------
>
> In index.php
> <https://github.com/glidernet/ogn-ddb/pull/57#discussion_r1516529340>:
>
> >      if ($req->rowCount() == 1) {        // if device already registred
>          $result = $req->fetch();
>          if ($result['dev_userid'] == $_SESSION['user']) {
>              $upd = true;
>          }        // if owned by the user then update
>          else {
> -            $error = $lang['error_devexists'];
> +            $ttime=time()-31536000; //1 year;
>
> Can you please try to have this as a constant and not repeat the same
> value multiple time?
> You can also say this is 365 * 24 * 60 * 60 instead of putting this number
> a bit unreadable.
> ------------------------------
>
> In language/english.php
> <https://github.com/glidernet/ogn-ddb/pull/57#discussion_r1518692985>:
>
> >      "table_update" => "Update",
>      "table_delete" => "Delete",
>      "table_new" => "Add new",
> +    "table_tt_expiration" =>"Expiration Status: Green Valid, Red Expired",
>
> Please add space before the second string as for the others
> ------------------------------
>
> In index.php
> <https://github.com/glidernet/ogn-ddb/pull/57#discussion_r1520155415>:
>
> > @@ -735,17 +748,25 @@ function devicelist()
>          fillindevice();
>      } else {
>          if ($upd) {
> -            $ins = $dbh->prepare('UPDATE devices SET dev_type=:dt, dev_actype=:ty, dev_acreg=:re, dev_accn=:cn, dev_notrack=:nt, dev_noident=:ni WHERE dev_id=:de AND dev_userid=:us');
> +            if ($trf) {
> +                $ins = $dbh->prepare('UPDATE devices SET dev_type=:dt, dev_actype=:ty, dev_acreg=:re, dev_accn=:cn, dev_notrack=:nt, dev_noident=:ni, dev_updatetime=:ti, dev_userid=:us WHERE dev_id=:de');
> +                //Transfer expired device
> +            }
> +            else {
>
> Please put } else { //Transfer expired device on a single line.
> ------------------------------
>
> In index.php
> <https://github.com/glidernet/ogn-ddb/pull/57#discussion_r1520156185>:
>
> >          } else {
> -            $ins = $dbh->prepare('INSERT INTO devices (dev_id, dev_type, dev_actype, dev_acreg, dev_accn, dev_userid, dev_notrack, dev_noident) VALUES (:de, :dt, :ty, :re, :cn, :us, :nt, :ni)');
> +            $ins = $dbh->prepare('INSERT INTO devices (dev_id, dev_type, dev_actype, dev_acreg, dev_accn, dev_userid, dev_notrack, dev_noident,dev_updatetime) VALUES (:de, :dt, :ty, :re, :cn, :us, :nt, :ni, :ti)');
>
> sapece missing before dev_updatetime
> ------------------------------
>
> In language/czech.php
> <https://github.com/glidernet/ogn-ddb/pull/57#discussion_r1520157198>:
>
> >      "table_update" => "Aktualizace",
>      "table_delete" => "Výmaz",
>      "table_new" => "Přidat nové",
> +    "table_tt_expiration" =>"Platnost registrace tohoto zařízení vyprší:",
> +    "expiry_disclaimer" =>"Zařízení s vypršenou registrací si mohou ostatní uživatelé zdarma zaregistrovat, obnovit zařízení s prošlou platností na další rok, upravit a ověřit své zařízení, zkontrolovat data, zkontrolovat certifikaci vlastnictví zařízení a stisknout Odeslat",
> +    "expiry_warning" =>"Upozornění: Platnost registrace některých vašich zařízení vypršela, zkontrolujte je",
>
> Sapce missing before "Upozornění
> ------------------------------
>
> In language/dutch.php
> <https://github.com/glidernet/ogn-ddb/pull/57#discussion_r1520157450>:
>
> >      "table_noident" => "Ident.",
>      "table_update" => "Update",
>      "table_delete" => "Verwijder",
>      "table_new" => "Toevoegen",
> +    "table_tt_expiration" =>"Deze apparaatregistratie verloopt na de volgende periode:",
> +    "expiry_disclaimer" =>"Apparaten met een verlopen registratie kunnen gratis door andere gebruikers worden geregistreerd, een verlopen apparaat met nog een jaar verlengen, uw apparaat bewerken en valideren, de gegevens bekijken, de eigendomscertificering van het apparaat controleren en op Verzenden drukken",
> +    "expiry_warning" =>"Waarschuwing: sommige van uw apparaatregistraties zijn VERLOPEN. Controleer ze alstublieft",
>
> again sapce missing
> ------------------------------
>
> In language/english.php
> <https://github.com/glidernet/ogn-ddb/pull/57#discussion_r1520157657>:
>
> >      "table_update" => "Update",
>      "table_delete" => "Delete",
>      "table_new" => "Add new",
> +    "table_tt_expiration" =>"This device registration is going to expire the:",
> +    "expiry_disclaimer" =>"Devices with expired registration are free to be registered by other users, to renew for another year an expired device, edit and validate your device, review the data, check the device ownership certification and press submit",
> +    "expiry_warning" =>"Warning: Some of your devices registration are EXPIRED, please review them",
>
> again
> ------------------------------
>
> In language/french.php
> <https://github.com/glidernet/ogn-ddb/pull/57#discussion_r1520157898>:
>
> >      "table_update" => "Modifier",
>      "table_delete" => "Supprimer",
>      "table_new" => "Ajouter",
> +    "table_tt_expiration" =>"L'enregistrement de cet appareil va expirer :",
> +    "expiry_disclaimer" =>"Les appareils dont l'enregistrement a expiré peuvent être enregistrés librement par d'autres utilisateurs, renouveler pour une année supplémentaire un appareil expiré, modifier et valider votre appareil, examiner les données, vérifier la certification de propriété de l'appareil et appuyer sur Soumettre.",
> +    "expiry_warning" =>"Avertissement : L'enregistrement de certains de vos appareils est EXPIRÉ, veuillez les vérifier",
>
> again
> ------------------------------
>
> In language/german.php
> <https://github.com/glidernet/ogn-ddb/pull/57#discussion_r1520158168>:
>
> >      "table_update" => "Aktualisieren",
>      "table_delete" => "Löschen",
>      "table_new" => "Hinzufügen",
> +    "table_tt_expiration" =>"Diese Geräteregistrierung läuft ab:",
> +    "expiry_disclaimer" =>"Geräte mit abgelaufener Registrierung können von anderen Benutzern kostenlos registriert werden, ein abgelaufenes Gerät um ein weiteres Jahr verlängern, Ihr Gerät bearbeiten und validieren, die Daten überprüfen, die Eigentumsbescheinigung des Geräts überprüfen und auf Senden klicken",
> +    "expiry_warning" =>"Warnung: Einige Ihrer Geräteregistrierungen sind ABGELAUFEN. Bitte überprüfen Sie sie",
>
> again
> ------------------------------
>
> In templates/devicelist.html.twig
> <https://github.com/glidernet/ogn-ddb/pull/57#discussion_r1520160241>:
>
> >          <tr>
>              <td>
>                  <a href="/?a=updatedev&devid={{ device.dev_id }}">
>                      <img src="/pict/mod.gif" title="{{ lang.table_update }}" class="devbutton">
>                  </a>
>              </td>
> +            <TD class="tab2"><iMG SRC="/pict/ex{{ device.dev_expired }}.png" title="{{ lang.table_tt_expiration }} {{ (device.dev_updatetime+expirationdelta)|date('j F, Y') }} "></TD>
>
> Please write TD il lower case as all others
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glidernet/ogn-ddb/pull/57#pullrequestreview-1923037777>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL3TFMI3CCRLZ645BICKM7TYYMZHLAVCNFSM6AAAAABELKUZFCVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSMRTGAZTONZXG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

snip commented:
Sorry for that! I thought you were seeing it as you reply on some of them. But it was maybe just my screenshot.

skizzofly commented:
Yeah its why i did not understand what you wanted :)
I fixed the screenshot issues three days ago

Il ven 15 mar 2024, 19:07 Sebastien Chaumontet ***@***.***>
ha scritto:

> Sorry for that! I thought you were seeing it as you reply on some of them.
> But it was maybe just my screenshot.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glidernet/ogn-ddb/pull/57#issuecomment-2000184659>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL3TFMOOZQYQPIAXLKYM5B3YYM2F3AVCNFSM6AAAAABELKUZFCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMBQGE4DINRVHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

skizzofly commented:
Hi with [e959115](https://github.com/glidernet/ogn-ddb/pull/57/commits/e9591157a3c5f9ebe55fb64c6d83a86b70e93215) i commited the changes requested, if it's all ok i will squash it 

Please note that in devicelist.html.twig the TD's in the first part of the tables are all lowercase, and in the second part of the table they are uppercase, i am not sure if there is a logic in this, anyway i did as you told me, lowering the case of my addon.

skizzofly reviewed (commented):
commited with [e959115](https://github.com/glidernet/ogn-ddb/pull/57/commits/e9591157a3c5f9ebe55fb64c6d83a86b70e93215)

snip commented on index.php:
```diff
@@ -20,6 +20,7 @@
 
 $url = 'https://ddb.glidernet.org/';
 $sender = 'contact@glidernet.org';
+const expirationdelta = 365*24*60*60;
```
Maybe add a short comment to explain this is the device registration expiration time in seconds?

snip replied:
And maybe we should add a bit more than 1year so this can be done during ARC renewal.
Something like 1y + 3m ?

skizzofly commented:
Seb, did you check my commits?
If they are OK I will squash the commits as usual..

snip commented:
> Seb, did you check my commits? If they are OK I will squash the commits as usual..

There is still 2 small reviews to fix.
Can you please check?

skizzofly commented:
Seb, if the fixes are regarding the comment and the 15 month i commited it with [aec3279](https://github.com/glidernet/ogn-ddb/pull/57/commits/aec3279c73346509546eef00ea2cc577c51f11ce)

![ogn_comment_fix](https://github.com/glidernet/ogn-ddb/assets/49754801/b05d2637-e961-4910-bf2f-eb1d445c7345)


IF the changes are other, i do not see them, so be sure to submit them..

snip commented:
I think just this one is still missing:
<img width="666" alt="image" src="https://github.com/glidernet/ogn-ddb/assets/466133/0577268a-cfa5-4e65-add2-53012b3a217f">

skizzofly commented:
fixed with [8ebe58f](https://github.com/glidernet/ogn-ddb/pull/57/commits/8ebe58fe7c4e7a08c95d4acb2623147be2ea54dc)
Check, if all is OK i will squash it up..
Ceers.

snip commented:
It seems we are now good!
Go for squash!

skizzofly commented:
squashed into [6d2f41b](https://github.com/glidernet/ogn-ddb/commit/6d2f41bf221f8d6d1ab1a666071296f5d98fd33e)

snip commented:
@skizzofly about the database_schema_upgrade.sql.txt can you please provide a version which randomize by 2 months the expiration date, so it will not be predictable (someone will not be able to try all device to get it for his own at the initial expiry time)?
Thanks
Everything else looks ok so far.

skizzofly commented:
Seb, the dev_updatetime is NOT the expiry date , but the timestamp that the device was updated.
But Seb, really you are telling me that there are people everyday trying to steal legittime device ID's so it's so VITAL to randomize the expiry date by two month? 
IMHO it's a useless waste of development time, as if the device is not free today i will try tomorrow, and if not after tomorrow, and if not in a week.. until it will be free.
Anyway I really have no idea how to randomize a number in mysql in a certain range..sorry for that.. :( 
Ale.

skizzofly commented:
As required.. generates some random numbers around the current time stamp.
If you have a better solution be free to implement it.. as mentioned before i am not so good in SQL.

[database_schema_upgrade.sql.txt](https://github.com/glidernet/ogn-ddb/files/14758360/database_schema_upgrade.sql.txt)

snip commented:
> But Seb, really you are telling me that there are people everyday trying to steal legittime device ID's so it's so VITAL to randomize the expiry date by two month? 

Not everyday, bit once this will be announced, some will see this and maybe put an alarm for the D day next year to be able to get lot of ownership.

snip commented:
> As required.. generates some random numbers around the current time stamp. If you have a better solution be free to implement it.. as mentioned before i am not so good in SQL.
> 
> [database_schema_upgrade.sql.txt](https://github.com/glidernet/ogn-ddb/files/14758360/database_schema_upgrade.sql.txt)

Thanks!

I'm going to try this.

flightenthusiast commented:
@skizzofly thank you for your contribution and what you are doing

skizzofly commented:
@flightenthusiast , maybe this will solve a long timed problem.. our club has such a problem with a glider registered in south africa.. the owner is disappeared and does not reply to mails and no one knows where he is , we tried to contact him but there is no trace of him.. they sold us his device leggily (via LXNav and we have the invoice) but still is registered.
With this mod problems like this may be solved automatically. 

@snip  try the script on a copy of the DB before going in production . ;) 

Ceers.

snip commented:
I just put it in production. Thanks a lot @skizzofly.

skizzofly commented:
Seb, I'm happy to have helped..
in some time i will also contribute to the Italian localization
Ceers.

On Mon, Apr 1, 2024 at 5:43 PM Sebastien Chaumontet <
***@***.***> wrote:

> I just put it in production. Thanks a lot @skizzofly
> <https://github.com/skizzofly>.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/glidernet/ogn-ddb/pull/57#issuecomment-2030021314>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AL3TFMMF3HMV6RGOMOOOWU3Y3F6CLAVCNFSM6AAAAABELKUZFCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMZQGAZDCMZRGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
